internal/cri/server/container_create_linux_test.go

```diff
@@ -1003,13 +1002,49 @@ func TestGenerateSeccompSecurityProfileSpecOpts(t *testing.T) {
 					ssp = csp
 				}
 				specOpts, err := cri.generateSeccompSpecOpts(ssp, test.privileged, !test.disable)
-				assert.Equal(t,
-					reflect.ValueOf(test.specOpts).Pointer(),
-					reflect.ValueOf(specOpts).Pointer())
 				if test.expectErr {
 					assert.Error(t, err)
 				} else {
 					assert.NoError(t, err)
+					if test.specOpts == nil && specOpts == nil {
+						return
+					}
+					if test.specOpts == nil || specOpts == nil {
+						t.Fatalf("unexpected nil specOpts, expected nil: %t, actual nil: %t", test.specOpts == nil, specOpts == nil)
+					}
+					// `specOpts` for seccomp only uses/modifies `*specs.Spec`, not
+					// `oci.Client` or `*containers.Container`, so let's construct a
+					// `*specs.Spec` and compare if the results are the same.
+					expected := runtimespec.Spec{
+						Linux: &runtimespec.Linux{},
+						Process: &runtimespec.Process{
+							Capabilities: &runtimespec.LinuxCapabilities{
+								Bounding: []string{
```

Comment:
I think we don't need this assignment here. It's kind of confusing. 
Anyway, we can do it in the follow-up.